src/methods/join.js

```diff
@@ -480,10 +480,12 @@ export function joinValuesAsConditional(
   condition: AbstractValue,
   v1: void | Value,
   v2: void | Value
-): AbstractValue {
+): Value {
   let result = AbstractValue.createFromConditionalOp(realm, condition, v1, v2);
-  if (v1) result.mightBeEmpty = v1.mightHaveBeenDeleted();
-  if (v2 && !result.mightBeEmpty) result.mightBeEmpty = v2.mightHaveBeenDeleted();
+  if (result instanceof AbstractValue) {
```

Comment:
Interesting and a bit surprising to see this treatment of the empty value. So our "conditional" is really something different than just the ternary JavaScript operator. I didn't realize that.

Reply:
createFromConditionalOp can now return concrete values and concrete values do not have the mightBeEmpty property. I introduced mightBeEmpty a long time ago to make it easier to model joins between branches where a property is missing and branches where it is present. We could possibly do the same for undefined and null, but since empty is not really a value, while the others are the analogy is somewhat strained. I'm still in a muddle about the whole thing and I'm putting it off until later.